### ISSUE TITLE
autoconfig.mydomain.com not working (error 500)

### ISSUE THREAD

**Issue by blackhold** [modoboa/modoboa-installer#475]
# Impacted versions

* Distribution: Debian
* Codename: Bullseye
* Arch: 64 Bits
* Database: PostgreSQL

# Steps to reproduce
http://autoconfig.mydomain.com/mail/config-v1.1.xml?emailaddress=admin@mydomain.com

# Full trace using --debug option or current behaviour

<pre>
root@mail:/var/log/uwsgi/app# tail -f automx_instance.log
2023-01-30 21:24:19,756 ERROR: data.configure(): Not an executable object: "SELECT first_name || ' ' || last_name AS display_name, email, SPLIT_PART(email, '@', 2) AS domain FROM core_user WHERE email='admin@mydomain.com' AND is_active"
Mon Jan 30 21:24:19 2023 - [ERROR] Unhandled object from iterator: '' (0x7feb0c8e7670)
autoconfig.mydomain.com [pid: 23530|app: 0|req: 2/4] 79.153.30.160 () {44 vars in 934 bytes} [Mon Jan 30 21:24:19 2023] GET /mail/config-v1.1.xml?emailaddress=admin@mydomain.com => generated 0 bytes in 239 msecs (HTTP/1.1 500) 2 headers in 81 bytes (1 switches on core 0)
</pre>

<pre>
modoboa=# SELECT first_name || ' ' || last_name AS display_name, email, SPLIT_PART(email, '@', 2) AS domain FROM core_user WHERE email='admin@mydomain.com' AND is_active;
 display_name |              email              |      domain      
--------------+---------------------------------+------------------
              | admin@mydomain.com | mydomain.com
(1 row)
</pre>

<pre>
root@mail:/etc/nginx/sites-enabled# su - automx
$ bash
automx@mail:~$ source env/bin/activate
(env) automx@mail:~$ automx-test admin@mydomain.com

Testing Autoconfig ...
Connecting to http://autoconfig.mydomain.com/mail/config-v1.1.xml?emailaddress=admin@mydomain.com ...

Trying fallback URL ...
Connecting to http://mydomain.com/.well-known/autoconfig/mail/config-v1.1.xml?emailaddress=admin@mydomain.com ...

No autoconfig endpoint found.

Testing Autodiscover (Microsoft Outlook(tm)) ...
Connecting to https://autodiscover.mydomain.com/autodiscover/autodiscover.xml ...

Testing Autodiscover (mobilesync) ...
Connecting to https://autodiscover.mydomain.com/autodiscover/autodiscover.xml ...

Testing mobileconfig...
Connecting to https://autodiscover.mydomain.com/mobileconfig ...
</pre>

Also an other thing is that, the fallback domain coult not be root domain, 'cause it goes to an other service not installed in the mail server. Does it is ok?

# Expected behavior

Appear XML configuration. I have an other server installed with old modoboa version that is working.

# Video/Screenshot link (optional)

**Comment by ShevT** [comment]
+1

**Comment by indiverbadal** [comment]
+1
Any help would be appreciated

**Comment by mbaechtold** [comment]
The problem is that `automx` is not compatible with SLQalchemy >= 2.0.

I have solved the problem by downgrading SQLalchemy: `sudo -u automx /srv/automx/env/bin/pip3 install "sqlalchemy < 2.0"`, followed by a reboot.

**Comment by blackhold** [comment]
> `sudo -u automx /srv/automx/env/bin/pip3 install "sqlalchemy < 2.0"`

many thanks! it worked! 

``` 
root@mail:/srv# sudo -u automx /srv/automx/env/bin/pip3 install "sqlalchemy < 2.0"
Collecting sqlalchemy<2.0
  Downloading SQLAlchemy-1.4.46-cp39-cp39-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (1.6 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.6/1.6 MB 38.9 MB/s eta 0:00:00
Requirement already satisfied: greenlet!=0.4.17 in ./automx/env/lib/python3.9/site-packages (from sqlalchemy<2.0) (2.0.2)
Installing collected packages: sqlalchemy
  Attempting uninstall: sqlalchemy
    Found existing installation: SQLAlchemy 2.0.0
    Uninstalling SQLAlchemy-2.0.0:
      Successfully uninstalled SQLAlchemy-2.0.0
Successfully installed sqlalchemy-1.4.46

[notice] A new release of pip available: 22.3.1 -> 23.0.1
[notice] To update, run: python3 -m pip install --upgrade pip

root@mail:/srv# reboot
```
